[Shawn Ng Hong Han] iP

snigloo opened the pull request:
### Welcome to my individual project!
> Anything is possible - Me 

Here are some things to take note:
- This is a WIP
- This is a project under CS2103T.

Make sure to:
1. _Count_ your **chickens.**
2. Test the bot using the jar file.

Here is some sample commands to use:
```
todo this
deadline homework /by 08/09/2022 1200
event birthday /at 08/09/2022 1400
```
🥳
Download the jar from [here](www.google.com).

- [x] downloaded the jar

Close the bot by commanding `bye`.

Commit:
Add Gradle support

snigloo committed:
Level-1

snigloo committed:
Level-2

snigloo committed:
Level-3

snigloo committed:
Level-4

snigloo committed:
Level-5

snigloo committed:
A-TextUiTesting

snigloo committed:
Level-6

snigloo committed:
Fixed delete

snigloo committed:
A-TextUiTesting

snigloo committed:
fixed bugs

snigloo committed:
OOP

snigloo committed:
Tasklist

snigloo committed:
Level-7

snigloo committed:
Level-8

snigloo committed:
Merge branch 'branch-Level-7'

snigloo committed:
fixed conflicts

snigloo committed:
Fixed spacing in save file

snigloo committed:
A-MoreOOP

snigloo committed:
A-Packages

snigloo committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

snigloo committed:
A-Gradle

snigloo committed:
A-JUnit

snigloo committed:
A-Jar

snigloo committed:
A-JavaDoc

snigloo committed:
A-CodingStandard

snigloo committed:
Level-9

snigloo committed:
Merge branch 'branch-A-JavaDoc'

snigloo committed:
Fixed merge conflict

snigloo committed:
Fixed merge conflicts

chengda300 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+                    ui.printDeleteMsg(removed.toString(), tl.size());
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("todo")) {
+                    if (curr.length() < 6) {
```
Perhaps using ```length()``` to express these numbers in accordance to the length of the command might be better so that the numbers do not look like "magic numbers".
It might also be better to check if the substring only contains spaces using the ```isBlank()``` method to determine if the task is empty.

This is also found for the subsequent lines of codes with "magic numbers".

chengda300 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+     * @param dateTime string with date and time in format (dd/mm/yyyy hhmm).
+     * @return LocalDateTime object containing date and time.
+     */
+    public static LocalDateTime dateParser(String dateTime) {
```
dateParser sounds more like a noun. Perhaps it is possible to make it sound more like a verb instead?

chengda300 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,137 @@
+    /**
+     * Prints goodbye message.
+     */
+    public void goodByeMessage() {
```
Perhaps making it sound more like a verb instead?

chengda300 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+    public static Task fileStringToTask(String task) {
+        String[] taskSplit = task.split("\\|");
+        String type = taskSplit[0];
+        switch (type) {
+            case TASK_TODO:
+                return new Todo(taskSplit[2].trim(), taskSplit[1].equals("1"));
+            case TASK_EVENT:
+                return new Event(taskSplit[2].trim(), taskSplit[3], taskSplit[1].equals("1"));
+            case TASK_DEADLINE:
+                return new Deadline(taskSplit[2].trim(), taskSplit[3], taskSplit[1].equals("1"));
+            default:
+                return null;
+        }
```
Based on coding standards, case statements do not need indentations.
Is it possible to remove the indentations?

```suggestion
        switch (type) {
        case TASK_TODO:
            return new Todo(taskSplit[2].trim(), taskSplit[1].equals("1"));
        case TASK_EVENT:
            return new Event(taskSplit[2].trim(), taskSplit[3], taskSplit[1].equals("1"));
        case TASK_DEADLINE:
            return new Deadline(taskSplit[2].trim(), taskSplit[3], taskSplit[1].equals("1"));
        default:
            return null;
        }
```

chengda300 reviewed (commented):
Overall, the code is quite well written other than the usage of magic numbers in parsing commands written to Duke. The parsing process is quite good though, detecting separate words instead of substrings as a whole.

I also like how local this bot sounds too. It is quite cute.

yiyefyy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("mark")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    if (index < 1 || index > tl.size()) {
```
Try to avoid deep nesting (more than 3 indentations), perhaps use switch to make your code cleaner :)

yiyefyy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+                    ui.printTaskAddedMsg(newTodo, tl.size());
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("deadline")) {
+                    if (curr.length() < 10 || !curr.contains("/by")) {
```
I think you can group the methods in like tasks or something. 
Eg.  In parser: task.addDeadline();
In Task: void addDeadline(){}

yiyefyy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+
+        ui.printIntro();
+
+        while (true) {
```
I think it will be better if you move this part to parser, like process the commands in paser. 😸

yiyefyy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+     * Writes tasks into storage file.
+     * @param tasklist TaskList to be written.
+     */
+    public static void listToFile(ArrayList<Task> tasklist) {
```
Maybe you can move this to storage

yiyefyy commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,22 @@
+public class EventTest {
+
+    @Test
+    public void newEventTest() {
```
You can try naming it as new_Event_Test (according to coding standards)

yiyefyy reviewed (commented):
Nice naming for booleans and methods! Code could be cleaner (move the methods to the correct files)

snigloo committed:
Fixed some coding standard violations

snigloo committed:
A-CheckStyle

snigloo committed:
Level-10

snigloo committed:
Level-10

snigloo committed:
No method descriptors for the addition of Basic GUI.

It is important to practice standard Java coding standards.

Method descriptors for all additional methods are being added at the moment.

Methods are created first, then the descriptors are added to not break workflow.

snigloo committed:
Messages need to be added for DialogBox class.

Change is needed to follow JavaDoc Coding Conventions.

Added JavaDocs for methods without it.

snigloo committed:
Assertions are being added to some of the Task classes.

It is important to check if the inputs are non-empty, and the TaskList not null.

Assert equals for strings and assert != null is being added as checks.

This is done to throw a AssertionError for major issues with changes to the code.

snigloo committed:
Made changes according to Code Quality

snigloo committed:
C-Help

snigloo committed:
Merge pull request #3 from snigloo/branch-A-Assertions

Assertions are being added to some of the Task classes.

snigloo committed:
Merge branch 'master' into branch-A-CodeQuality

snigloo committed:
Merge pull request #4 from snigloo/branch-A-CodeQuality

Made changes according to Code Quality

snigloo committed:
Made changes to Todo

snigloo committed:
Fixed merge conflicts for BCD-Extension

snigloo committed:
A-UserGuide

snigloo committed:
Updated ReadME

snigloo committed:
A-Release

snigloo committed:
Minor changes to name of jar file
